src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java

```diff
@@ -0,0 +1,1406 @@
+    {
+      double min_x = Util.px2mm(p.getMinX(), p.getDPI());
+      double min_y = Util.px2mm(p.getMinY(), p.getDPI());
+      double max_x = Util.px2mm(p.getMaxX(), p.getDPI());
+      double max_y = Util.px2mm(p.getMaxY(), p.getDPI());
+      if (first) {
+        minX = min_x;
+        maxX = max_x;
+        minY = min_y;
+        maxY = max_y;
+        first = false;
+      }
+      else {
+        if (min_x < minX) { minX = min_x; }
+        if (max_x > maxX) { maxX = max_x; }
+        if (min_y < minY) { minY = min_y; }
+        if (max_y > maxY) { maxY = max_y; }
+      }
+    }
+
+    /* bounding box */
+    stream.hex("E703").absoluteMM(minX).absoluteMM(minY);
+    stream.hex("E707").absoluteMM(maxX).absoluteMM(maxY);
+    stream.hex("E750").absoluteMM(minX).absoluteMM(minY);
+    stream.hex("E751").absoluteMM(maxX).absoluteMM(maxY);
+    stream.hex("E7040001000100000000000000000000");
+    stream.hex("E70500");
+  }
+
+
+  private double last_x = 0.0;
+  private double last_y = 0.0;
+  private int vector_count = 0;
+  private long travel_distance = 0;
+
+  private void vector(double x, double y, double dpi, boolean as_cut)
+  {
+    double x_mm = Util.px2mm(x, dpi);
+    double y_mm = Util.px2mm(y, dpi);
+    boolean as_absolute;
+
+    /* compute distance to last known position */
+    double dx = x_mm - last_x;
+    double dy = y_mm - last_y;
+
+    if ((dx == 0.0) && (dy == 0.0)) {
+      return;
+    }
+    if (vector_count % 10 == 0) {                  /* enforce absolute every 10 vectors */
+      as_absolute = true;
+    }
+    else {
+      as_absolute = Math.max(Math.abs(dx), Math.abs(dy)) > 8.191;
+    }
+    vector_count += 1;
+
+    long distance = (long)Math.sqrt(dx*dx + dy*dy);
+    travel_distance += distance;
+
+    // estimate the new real position
+    last_x = x_mm;
+    last_y = y_mm;
+
+    if (as_absolute) {
+      if (as_cut) {
+        stream.hex("A8").absoluteMM(x_mm).absoluteMM(y_mm);  // cut to x,y
+      }
+      else { // as_move
+        stream.hex("88").absoluteMM(x_mm).absoluteMM(y_mm);  // move to x,y
+      }
+    }
+    else { // relative
+      if (dx == 0.0) {
+        if (as_cut) {
+          stream.hex("AB").relativeSignedMM(dy);  // cut vertical to y
+        }
+        else { // as_move
+          stream.hex("8B").relativeSignedMM(dy);  // move vertical to y
+        }
+      }
+      else if (dy == 0.0) {
+        if (as_cut) {
+          stream.hex("AA").relativeSignedMM(dx);  // cut horizontal to y
+        }
+        else { // as_move
+          stream.hex("8A").relativeSignedMM(dx);  // move horizontal to y
+        }
+      }
+      else {
+        if (as_cut) {
+          stream.hex("A9").relativeSignedMM(dx).relativeSignedMM(dy); // cut relative to x, y
+        }
+        else { // as_move
+          stream.hex("89").relativeSignedMM(dx).relativeSignedMM(dy); // move relative to x, y
+        }
+      }
+    }
+  }
+
+  private float currentMinPower = 0.0f;
```

Comment:
should be "transient" (please also check all other variables that appear in the ~/.visicut/devices/....xml file and are not necessary for permanent storage, like vector_count, travel_distance, etc.)

Reply:
fixed for `currentMinPower`

Reply:
fixed for everything else